tests/utils/test_rest_utils.py

```diff
@@ -185,64 +196,34 @@ def test_ignore_tls_verification_not_server_cert_path():
         )
 
 
-@mock.patch("requests.request")
-def test_429_retries(request):
```

Comment:
Since we rely on the HTTPAdaptor and Retry function. This test is no longer necessary.

As a side note, it's also non-trivial to test the Retry functionality on the high level without an actual internet connection. https://github.com/getsentry/responses/issues/135

Reply:
@jinzhang21 Can we conduct a manual test for this change, perhaps on Databricks?

Reply:
Love to. What's the standard way of testing client change on Databricks? Build a wheel and manually install it?

Reply:
You got it! You can also install from this branch using `%pip install git+https://github.com/mlflow/mlflow@ML-16770`, which I find to be faster.

Reply:
Ran a E2E xgboost training process on a test shard. I was able to verify the timeout behavior but couldn't verify the retry logic unfortunately. What's the best way to simulate a temporary error on Databricks backend? When I try to throw an exception on the backend, it registers as an error (permanent) on the RPC console. Not sure what error code / response was actually returned to the client.

Reply:
Update. This is because Retry does not work by default on POST method: https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.Retry.DEFAULT_ALLOWED_METHODS

Updated the param to None to allow all HTTP methods to be retried. Verified the retry works as expected on test shard.